pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BrokerUserDefinedMessageHandlerFactory.java

```diff
@@ -129,6 +135,32 @@ public void onError(Exception e, ErrorCode code, ErrorType type) {
     }
   }
 
+  private class RefreshDatabaseConfigMessageHandler extends MessageHandler {
+    final String _databaseName;
+
+    RefreshDatabaseConfigMessageHandler(DatabaseConfigRefreshMessage databaseConfigRefreshMessage,
+        NotificationContext context) {
+      super(databaseConfigRefreshMessage, context);
+      _databaseName = databaseConfigRefreshMessage.getDatabaseName();
+    }
+
+    @Override
+    public HelixTaskResult handleMessage() {
+      // only update the existing rate limiter.
+      // Database rate limiter creation should only be done through table based change triggers
+      _queryQuotaManager.updateDatabaseRateLimiter(_databaseName);
```

Comment:
Can this cause race conditions between table creation and database quota config updates resulting in a stale quota config being used?

Reply:
That's a good question. The `ClusterChangeMediator` has a single thread which process these changes so there should not be a race condition. Will double check on this.

Reply:
I'd assume custom HelixMsg would be processed by a separate thread pool, while the ClusterChangeMediator may have its own internal thread pool to invoke listeners upon Helix changes. can confirm by printing the thread names

Reply:
synchronized `createOrUpdateDatabaseRateLimiter` to avoid the race condition on rate limiter update